src/common/sockets.cc

```diff
@@ -632,11 +636,15 @@ int32_t tcptowrite(int sock, const void *buff, uint32_t leng, int msecto) {
 	while (sent < leng) {
 		pfd.revents = 0;
 		if (tcppoll(pfd, msecto) < 0) {
+			tcpsetlasterror(TCPETIMEDOUT);
 			return -1;
 		}
```

Comment:
From `man poll` (again, this just seems to be wrapper function to poll, but the `npfd` parameter hard coded to 1):

```
RETURN VALUE
       Upon successful completion, poll() shall return a non-negative value.  A
       positive value indicates the total number of pollfd structures that have
       selected  events  (that  is,  those for which the revents member is non-
       zero). A value of 0 indicates that the call timed out and  no  file  de‐
       scriptors  have  been selected. Upon failure, poll() shall return -1 and
       set errno to indicate the error.

ERRORS
       The poll() function shall fail if:

       EAGAIN The allocation of internal data structures failed  but  a  subse‐
              quent request may succeed.

       EINTR  A signal was caught during poll().

       EINVAL The  nfds  argument  is greater than {OPEN_MAX}, or one of the fd
              members refers to a STREAM or multiplexer  that  is  linked  (di‐
              rectly or indirectly) downstream from a multiplexer.
```

Judging the possible errors, the previous code is correct that the only thing left to do in this case is return failure with `-1`. Setting the error value is wrong here because it hides the actual error that happened and this timeout is covered in a case below.

Reply:
Decided to delete all code affected from common and let error management for mount.